### PR TITLE
[OpenMP] Improve omp offload profiler

### PR DESCRIPTION
**Pull request by fel-cab**
Summary:
Adding information to the LIBOMPTARGET profiler runtime kernel and API calls.

Key changes:
* Adding information to runtime calls for better understanding of how the application
  is executing. For example teams requested by the user, size of memory transfers.
* Profile timer was changed from 'us' to 'ns', since 'us' was too coarse-grain
  to register some important details like key kernel duration
* Removed non API or Runtime calls, to reduce complexity of profile for application
  developers.

### CODE REVIEW

File: llvm/lib/Support/TimeProfiler.cpp [hunk 1]
```diff
@@ -79,15 +79,15 @@ struct TimeTraceProfilerEntry {
   // Calculate timings for FlameGraph. Cast time points to microsecond precision
```
Comment (jdoerfert) [inline llvm/lib/Support/TimeProfiler.cpp:79]: Can you split the changes to this file into a separate PR? Also check the file, the comment still states microseconds.

Comment (fel-cab) [reply llvm/lib/Support/TimeProfiler.cpp]: Removed this file from this PR

File: openmp/libomptarget/src/private.h [hunk 2]
```diff
@@ -432,12 +432,21 @@ class ExponentialBackoff {
 #define TIMESCOPE_WITH_RTM_AND_IDENT(RegionTypeMsg, IDENT)                     \
   SourceInfo SI(IDENT);                                                        \
   std::string ProfileLocation = SI.getProfileLocation();                       \
+  std::string ProfileName = SI.getName();                                      \
   std::string RTM = RegionTypeMsg;                                             \
   llvm::TimeTraceScope TimeScope(__FUNCTION__, ProfileLocation + RTM)
+// llvm::TimeTraceScope TimeScope(RegionTypeMsg, ProfileLocation + RTM)
```
Comment (markdewing) [inline openmp/libomptarget/src/private.h:438]: minor: commented-out code

**Review by jdoerfert — APPROVED** [review]
LG. Please rebase and merge.

**Pull request merged**